src/proto/faabric.proto

```diff
@@ -93,12 +96,10 @@ message MPIMessage {
 }
 
 message Message {
-    string user = 1;
-    string function = 2;
-    string snapshotKey = 4;
-    int32 snapshotSize = 5;
-    int32 funcPtr = 6;
-    int32 returnValue = 7;
+    int32 id = 1;
+    int32 appId = 2;
+    int32 appIndex = 3;
+    string masterHost = 4;
```

Comment:
I introduced these fields, but rearranged and renumbered the rest as they didn't make sense. According to the protobuf docs it's also most efficient to use numbers 1-15 for regularly used fields, so I made sure that's the case.